### ISSUE TITLE
False positive of `no-undefined-references` on YAML frontmatter

### ISSUE THREAD

**Issue by ghost** [remarkjs/remark-lint#275]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

remark-preset-lint-recommended@6.1.1 remark-cli@10.0.0 remark-lint@9.1.0

### Steps to reproduce

Simply run `remark` on any markdown that contains a frontmatter array, example below:

```markdown
---
tags: ['lorem', 'ipsum', 'foo', 'bar']
---

# Hello World!
```

### Expected behavior

Should still keep warning of possible errors, without taking into consideration the frontmatter YAML

### Actual behavior

Flagging the frontmatter YAML array as a undefined reference

### Runtime

Node v16

### Package manager

npm v7

### OS

Linux

### Build and bundle tools

Snowpack
